test/test.js

```diff
@@ -164,6 +164,44 @@ function behavesLikeParseObjectOnBeforeDelete(typeName, ParseObjectOrUserSubclas
   });
 }
 
+function behavesLikeParseObjectOnAfterSave(typeName, ParseObjectOrUserSubclass) {
+  context('when object has afterSave hook registered', () => {
+    const errorMessage = 'Error in afterSave hook test';
+    let didAfterSave = false;
+    function afterSavePromise(request) {
+      if (request.object.get('error')) {
+        return Parse.Promise.error(errorMessage);
+      }
+      didAfterSave = true;
+      return Parse.Promise.as();
+    }
+
+    it('runs the hook after saving the model and persists the object', () => {
+      ParseMockDB.registerHook(typeName, 'afterSave', afterSavePromise);
+
+      const object = new ParseObjectOrUserSubclass();
+      assert(!object.has('cool'));
```

Comment:
I think you should set `didAfterSave = false` here since it's mutable, just to be certain that it does change between here and the callback through the afterSave hook. And prevent any false positives in the future if someone adds a test and decides for some reason to mess around with `didAfterSave`.